File: android/app/src/main/java/com/tandem/emt/features/incidentlist/models/IncidentModels.kt
package com.tandem.emt.features.incidentlist.models

enum class IncidentType { FIRE, EMS, HAZMAT, RESCUE, OTHER }

enum class Priority { HIGH, MEDIUM, LOW }

enum class IncidentStatus {
    DISPATCHED, EN_ROUTE, ON_SCENE, CLEARED;

    val isActive: Boolean get() = this != CLEARED
}

data class UnitSummary(val id: String, val name: String)

data class Coordinates(val latitude: Double, val longitude: Double)

data class IncidentSummary(
    val id: String,
    val caseNumber: String,
    val type: IncidentType,
    val priority: Priority,
    val address: String,
    val dispatchTime: Long,
    val status: IncidentStatus,
    val assignedUnits: List<UnitSummary>,
    val coordinates: Coordinates
)

enum class FilterStatus(val displayName: String) {
    ALL("All"),
    ACTIVE("Active"),
    MY_ASSIGNED("My Assigned")
}

File: ios/TandemEMT/Features/IncidentList/Models/IncidentModels.swift
import Foundation

enum IncidentType: String, Codable, CaseIterable {
    case fire, ems, hazmat, rescue, other
}

enum Priority: String, Codable {
    case high, medium, low
}

enum IncidentStatus: String, Codable {
    case dispatched, enRoute, onScene, cleared

    var isActive: Bool {
        self != .cleared
    }
}

struct UnitSummary: Codable, Identifiable, Equatable {
    let id: String
    let name: String
}

struct Coordinates: Codable, Equatable {
    let latitude: Double
    let longitude: Double
}

struct IncidentSummary: Codable, Identifiable, Equatable {
    let id: String
    let caseNumber: String
    let type: IncidentType
    let priority: Priority
    let address: String
    let dispatchTime: Date
    let status: IncidentStatus
    let assignedUnits: [UnitSummary]
    let coordinates: Coordinates
}

enum FilterStatus: String, CaseIterable {
    case all, active, myAssigned

    var displayName: String {
        switch self {
        case .all: return "All"
        case .active: return "Active"
        case .myAssigned: return "My Assigned"
        }
    }
}
